core/src/main/scala/kafka/tools/MirrorMaker.scala

```diff
@@ -527,18 +527,18 @@ object MirrorMaker extends Logging with KafkaMetricsGroup {
       }
     }
 
-    // New consumer always hasNext
     override def hasData = true
 
     override def receive() : BaseConsumerRecord = {
-      while (recordIter == null || !recordIter.hasNext)
+      if (recordIter == null || !recordIter.hasNext) {
         recordIter = consumer.poll(1000).iterator
+        if (!recordIter.hasNext)
+          throw new ConsumerTimeoutException
+      }
 
-      val record = recordIter.next()
+      val record = recordIter.next
```

Comment:
The `()` should actually remain there since this is a side-effect (I know that you simply copied the code from `NewShinyConsumer`.